news/_posts/2019-09-11-prestashop-in-2019-and-beyond-part-2-pain-points.md

```diff
@@ -0,0 +1,471 @@
+
+In the meantime, we will continue adding VueJS in the Back Office, but only as standalone components like the [live search engine results preview component][serp] that we added in the Product Page in 1.7.5.0.
+
+
+### Duplicate systems
+
+Many of the new systems in PrestaShop 1.7 don't replace the previous ones, but feature a new, semi-independent way of doing the same thing that coexists with the previous way—which is kept for backwards compatibility. 
+
+Having more than one way of doing a given thing is bad for many reasons. First, it introduces cognitive friction, because as a developer you have to think about in when you should use one or the other. Second, because a business or dependency change in one must be replicated in the other. And third, if they don't produce the same output or side-effects for some reason, then it might produce systemwide inconsistencies.
+
+There are several examples of this in PrestaShop 1.7. One of them is the "Core–Adapter–Legacy" architecture. Let's explore four problems tied to this example:
+
+**1. Shared responsibility**
+	
+As described in the previous article, in order to allow old code to progressively be replaced by new code, the Core namespace conventions prohibits developers from using legacy classes directly in Core classes. When developers need legacy functionality in a Core class, it needs to be wrapped in an Adapter class that can then be injected into the Core class.
+	
+**The problem:** Since all the Adapter does is wrap legacy features in a non-static class, we now have two classes that do the same: one in Adapter, one in legacy. Overall complexity has increased.
+
+**2. Different implementations/side effects**
+
+The idea behind Adapters is that they should eventually be replaced by Core code that reimplements the feature, allowing us to delete the legacy class later. However, some developers chose to partially reimplement some legacy features in Adapters, but introducing a _slightly_ different behavior.
+	
+**The problem:** Not only do we have two classes that do the same thing, but they do it in a _different_ way and produce _different_ side effects. Which one should we use, and when?
+
+**3. Little or no added value**
+
+Looking at the code, we can see that many hastily made Adapters have been designed [as dumb bridges to legacy code][bad-example], even replicating method signatures and design flaws of their legacy counterparts. Some even have static methods!
+
+**The problem:** Many adapters do not even add any value to legacy classes and exist only to comply with the "no use of legacy classes in new code".
+
+**4. Hard dependencies**
+
+To make matters worse, most Adapters do not even implement an interface, which makes it impossible to replace them with Core reimplementations without modifying the dependent classes.
```

Comment:
```suggestion
To make matters worse, most Adapters do not even implement an [interface](https://www.php.net/manual/en/language.oop5.interfaces.php), which makes it impossible to replace them with Core reimplementations without modifying the dependent classes.
```

As you have talked about other interfaces before (API, contracts) maybe it's good to be accurate here when using the word